pynumaflow/mapper/multiproc_server.py

```diff
@@ -148,46 +147,57 @@ def _run_server(self, bind_address: str) -> None:
         map_pb2_grpc.add_MapServicer_to_server(self, server)
         server.add_insecure_port(bind_address)
         server.start()
-        serv_info = ServerInfo(
-            protocol=Protocol.TCP,
-            language=Language.PYTHON,
-            version=get_sdk_version(),
-            metadata=get_metadata_env(envs=METADATA_ENVS),
-        )
-        # Overwrite the CPU_LIMIT metadata using user input
-        serv_info.metadata["CPU_LIMIT"] = str(self._process_count)
-        info_server_write(server_info=serv_info, info_file=SERVER_INFO_FILE_PATH)
-
         _LOGGER.info("GRPC Multi-Processor Server listening on: %s %d", bind_address, os.getpid())
         server.wait_for_termination()
 
     @contextlib.contextmanager
-    def _reserve_port(self) -> int:
+    def _reserve_port(self, port_num: int) -> int:
```

Comment:
```suggestion
    def _reserve_port(self, port_num: int) -> Iterator[int]:
```
Since this is a generator function, and does not have a sendtype and returntype. 
https://docs.python.org/3.11/library/typing.html#typing.Generator